src/com/google/enterprise/adaptor/DocIdSender.java

```diff
@@ -461,7 +461,7 @@ private String alternateGroupSource(String source) {
       fileArchiver.saveFeed(feedSourceName, groupsDefXml);
       journal.recordGroupPush(defs);
     } else {
-      last = defs.get(0).getKey();  // checked in pushGroupDefinitionsInternal()
+      last = defs.isEmpty() ? null : defs.get(0).getKey();
```

Comment:
Setting last to null here mimics success and only works in pushGroupDefinitionsInternal because the batch is empty. I think that's worth a comment.